core/src/main/scala/com/github/gpoirier/barrage/actions.scala

```diff
@@ -60,11 +61,20 @@ object actions {
       object Three extends Workshop(3)
     }
 
-    sealed trait MachineShop extends Action
+    case class MachineShop(machinery: Machinery) extends Action {
+      def cost(column: ActionColumn): Cost = this -> column match {
+        case (MachineShop.Excavator, ActionColumn.Cheap) => Cost(1.eng, 2.credit)
+        case (MachineShop.Excavator, ActionColumn.Expensive) => Cost(1.eng, 5.credit)
+        case (MachineShop.Wild, ActionColumn.Cheap) => Cost(1.eng, 4.credit)
+        case (MachineShop.Wild, ActionColumn.Expensive) => Cost(2.eng, 4.credit)
+        case (MachineShop.Both, ActionColumn.Cheap) => Cost(2.eng, 5.credit)
+        case (MachineShop.Both, ActionColumn.Expensive) => Cost(3.eng, 8.credit)
+      }
+    }
     object MachineShop {
-      case object Excavator extends MachineShop
-      case object Wild extends MachineShop
-      case object Both extends MachineShop
+      object Excavator extends MachineShop(Machinery(1.excavator))
+      object Wild extends MachineShop(Machinery(1.excavator)) // not sure how to handle this..
```

Comment:
The intent was to have `MachineShop` contain a reward instead of a resource instance, so that here could contain the value `WildMachinery(1)`.

Reply:
Yeah, I just couldn't figure out how to handle that.  I'm wondering if we really need a Reward.WildMachinery?  Maybe we only need a Reward.Machinery, because we can expect the user/parser to have already declared which machinery they are taking with the wild by the time the reward is resolved.

Reply:
That's not how the Command / Action separation works in the current model though